seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSourceSplitEnumerator.java

```diff
@@ -38,16 +42,28 @@ public class JdbcSourceSplitEnumerator implements SourceSplitEnumerator<JdbcSour
     private static final Logger LOG = LoggerFactory.getLogger(JdbcSourceSplitEnumerator.class);
     private final SourceSplitEnumerator.Context<JdbcSourceSplit> enumeratorContext;
 
-    private final Map<Integer, Set<JdbcSourceSplit>> pendingSplits;
+    private final Map<Integer, List<JdbcSourceSplit>> pendingSplits;
+
+    private final Object stateLock = new Object();
+    private volatile boolean shouldEnumerate;
 
     private JdbcSourceOptions jdbcSourceOptions;
     private final PartitionParameter partitionParameter;
 
     public JdbcSourceSplitEnumerator(SourceSplitEnumerator.Context<JdbcSourceSplit> enumeratorContext, JdbcSourceOptions jdbcSourceOptions, PartitionParameter partitionParameter) {
+        this(enumeratorContext, jdbcSourceOptions, partitionParameter, null);
+    }
+
+    public JdbcSourceSplitEnumerator(SourceSplitEnumerator.Context<JdbcSourceSplit> enumeratorContext, JdbcSourceOptions jdbcSourceOptions, PartitionParameter partitionParameter, JdbcSourceState sourceState) {
         this.enumeratorContext = enumeratorContext;
         this.jdbcSourceOptions = jdbcSourceOptions;
         this.partitionParameter = partitionParameter;
         this.pendingSplits = new HashMap<>();
+        this.shouldEnumerate = sourceState == null;
```

Comment:
```suggestion
        if (sourceState == null) {
            this.shouldEnumerate = true;
        } else {
            this.shouldEnumerate = false;
        }
```

Reply:
```suggestion
        this.shouldEnumerate = sourceState == null ? true : false;
```
How about using this way?